src/geometry/types.rs

```diff
@@ -172,6 +172,21 @@ impl Rect {
         }
     }
 
+    // TODO: needs testing
+    // TODO: implement PartialOrd for point
+    pub fn contains(&self, point: &Point) -> bool {
+        point.x >= self.origin.x && point.y >= self.origin.y &&
```

Comment:
can point.x also < than ` self.origin.x && point.y < self.origin.y && point.x <= self.upper_bound().x && point.y <= self.upper_bound().y`

Reply:
NO because if the point is less than the origin, then it is not contained in the rect